### ISSUE TITLE
Consider not to use key in tr.b.Settings.set when key is undefined

### ISSUE THREAD

**Issue by lpy** [catapult-project/catapult#2889]
When use `tr.ui.b.createCheckBox` I don't want to use settingsKey, but even when I pass in a `undefined`, it still treat it as a valid key.

My question is:

1. Is there a way to avoid this currently in the codebase?
2. If no, should we make [`set`](https://github.com/catapult-project/catapult/blob/master/tracing/tracing/base/settings.html#L79) ignore `undefined`?

**Comment by lpy** [comment]
cc @zeptonaut @natduca

**Comment by benshayden** [comment]
What's the context? Why don't you want to remember the user's preference for the checkbox?

**Comment by lpy** [comment]
because I am creating a bunch of checkboxes in a loop based on selected slices, and they will trigger the same callback when they get changed.

In this case, I always want these checkboxes unchecked.

**Comment by benshayden** [comment]
Good reason. :-)
The [no defensive programming](https://github.com/catapult-project/catapult/blob/master/docs/style-guide.md#avoid-defensive-programming-and-document-it-when-you-cant) recommendation would seem to argue for having dom_helpers avoid calling tr.b.Settings when settingsKey === undefined, instead of having Settings tolerate undefined keys. Settings has one job, callers should only call it when they want that job done.
WDYT?

**Comment by lpy** [comment]
I completely agree. I actually end up handling `undefined` in dom_helpers. I will change the title.
